frontend/hooks/useAutosavedState.ts

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+
+export default function useAutosavedState<T>(
+  initialValue: T,
+  opts = { key: 'default', debounceTime: 1000 },
+): [T, (value: T) => void, () => void] {
+  const storage: any = (typeof window !== 'undefined' && window.localStorage) || {}
```

Comment:
Could there be a mismatch where SSR starts out with the empty object literal, and then `localStorage` is used on the client?

Reply:
This should get re-evaluated on every render, so if it gets SSR'd, then handed off to the browser, it should just switch to using local storage. Reloading the page on browser (so doing an SSR) it seemed to hang onto my content without any issue.

Reply:
Oh, and that behavior would be desirable. Since there's not really anything to be done here during SSR (we don't have this content on the server) just using an empty object and not restoring anything is what we want.